Revice Privacy Considerations (#130)
ietf-rats-wg/architecture: draft-ietf-rats-architecture.md

# Privacy Considerations

> Every claim in Attestation Evidence and Attestation Results is potentially PII (Personally Identifying Information) depending on the end-end use case of the attestation.

Reviewer: ```suggestion
Every Claim in attestation Evidence and Attestation Results is potentially PII (Personally Identifying Information) depending on the end-end use case of the remote attestation procedure.
```

Reviewer 2: I don't think it's true that every claim is PII.  If someone adds a claim saying it supports a specific version of the EAT spec, just like every other device, I wouldn't call that PII.  Similarly if there is a claim that contains the current time, or a nonce supplied by a verifier, those wouldn't be PII either under any use case I can think of.   The key question is whether it significantly narrows the anonymity set (as discussed in RFC 6973).

Reviewer 2: I also think the original text in red needs to stay at the top.  PII is not the only aspect of privacy, it's just an important subset.

Reviewer 3: Strike "Every", change "Claim" to "Claims" and change "is" to "are". Then I'm OK with the proposed new text.

> Each deployment of attestation must take into account how each claim may or may not be used in a privacy violating manner by the relying parties receiving the attestation.

Reviewer: ```suggestion
Each deployment of remote attestation must take into account how each Claim may or may not be used in a privacy violating manner by Verifiers receiving Evidence or Relying Parties receiving Attestation Results.
```

Reviewer 3: The text after "Verifiers" and "Relying Parties" is redundant / unnecessary since we explain in other sections the function of Verifiers and Relying Parties. Strike "receiving Evidence" and "receiving Attestation Results"

> In cases where the device sending attestation is owned by the relying party, little to no consideration for privacy is needed. In other cases, such as when a mobile phone is used to access a large number of web sites, apps and services, privacy must be considered very thoroughly.

Reviewer: ```suggestion
In cases where the Attester sending Evidence is owned by the Relying Party Owner, little to no consideration for privacy is needed. In other cases, such as when a mobile phone is used to access a large number of web sites, apps and services, privacy must be considered very thoroughly.
```

Reviewer: Not entirely sure that this suggested change is not breaking the intent of the original text.

Reviewer 3: "In cases where Evidence is owned by the Relying Party Owner, consideration for privacy may be of lesser importance. In other cases where Evidence is owned by the Attester, such as when a mobile phone is used to access a large number of web sites, applications or services, consideration for privacy may be of greater importance."

These are some strategies that can be used to address the privacy issue:

* Determine that a particular claim is not privacy violating in the limited context of use.
  For example, a router sending its serial number to the network management center that owns the router violates no one’s privacy.

* Simply omit the claim. For example, omit the serial number from attestations sent by a phone to web sites it connects to.

* Prompt the user. For example, ask the user if it is OK for a particular web site to access location data.

> * Use a privacy proxy service between the device and the relying party to remove or anonymize the privacy violating claims.

Reviewer 3: change 'device' to attester to be consistent use of roles terminology

Confidentiality will usually be necessary when conveying Attestation Evidence and Results so attackers can't capture privacy violating claims or gain information that makes attacking the device easier.
Evidence and Attestation Results data structures are expected to support integrity protection encoding (e.g., COSE, JOSE, X.509) and optionally might support confidentiality protection (e.g., COSE, JOSE).
If confidentiality protection is omitted or unavailable, the protocols that convey Evidence or Attestation Results should provide confidentiality.

The Verifier may or may not be trusted with privacy sensitive claims depending on the use case and nature of the Verifier.
For example, if the Verifier is operated by the Relying Party and that Relying Party should not be allowed to access to privacy violating claims, those claims must not be in Attestation Evidence.
On the other hand, the Verifier might be designed and operated to be a privacy proxy in which case part of its job is to remove or anonymize privacy violating claims in Attestation Evidence so they are not passed on in Attestation Results.

> Thus, in use cases where privacy violating claims are in the Attestation Evidence, the Attester should authenticate the Verifier to know that it is trusted. 

Reviewer: ```suggestion
Thus, in usage scenarios where privacy violating Claims are in the attestation Evidence, an Attester should authenticate Verifiers to know that they are trusted. 
```

Reviewer 3: Maybe avoid using "trusted" and instead say "Verifiers that have acceptable privacy policies".
